pyccl/base/schema.py

```diff
@@ -328,7 +336,57 @@ def __init__(self, a=1, b=2, c=3, d=4, e=5):
 
     def __repr__(self):
         # Build string from specified `__repr_attrs__` or use Python's default.
+        # Subclasses overriding `__repr__`, stop using `__repr_attrs__`.
         if hasattr(self.__class__, "__repr_attrs__"):
             from .repr_ import build_string_from_attrs
             return build_string_from_attrs(self)
         return object.__repr__(self)
+
+
+def _subclasses(cls):
+    # This helper returns a set of all subclasses.
+    direct_subs = cls.__subclasses__()
+    deep_subs = [sub for cl in direct_subs for sub in cl._subclasses()]
+    return set(direct_subs).union(deep_subs)
+
+
+def from_name(cls, name):
+    """Obtain particular model."""
+    mod = {p.name: p for p in cls._subclasses() if hasattr(p, "name")}
+    return mod[name]
+
+
+def create_instance(cls, input_, **kwargs):
+    """Process the input and generate an object of the class.
+    Input can be an instance of the class, or a name string.
+    Optional ``**kwargs`` may be passed.
+    """
+    if isinstance(input_, cls):
+        return input_
+    if isinstance(input_, str):
+        class_ = cls.from_name(input_)
+        return class_(**kwargs)
+    good, bad = cls.__name__, input_.__class__.__name__
+    raise TypeError(f"Expected {good} or str but received {bad}.")
```

Comment:
Why not define all these directly in `CCLNamedClass`?

Reply:
You have to *assign* them to the subclasses. It is kind of like binding, but for classes. If you defined it in the class and let it inherit, then `cls` would only refer to the absolute superclass of the entire tree, which is not what we want.

Reply:
I can define them in the class body but I'd still do the assignment operations, if that's what you prefer.

Reply:
There is no 1065. What do you want to do here? I think I'd rather you do this here, so everything is clean.

Reply:
1068.